gaitmap_mad/gaitmap_mad/event_detection/_rampp_event_detection.py

```diff
@@ -277,8 +480,32 @@ def _detect_ic(
     return float(acc_search_region_start + np.argmin(acc_pa_inv[acc_search_region_start:acc_search_region_end]))
 
 
-def _detect_tc(gyr_ml: np.ndarray) -> float:
+def _detect_tc_for_segmented_stride(gyr_ml: np.ndarray) -> float:
     try:
         return np.where(np.diff(np.signbit(gyr_ml)))[0][0]
     except IndexError:
         return np.nan
+
+
+def _find_first_negative_before_peak(arr):
+    """To find the tc.
+
+    find the peak of the signal and then search for the first negative value before the peak
+    """
+    peak_index = np.argmax(arr)  # Find the index of the largest positive number (peak)
+
+    # Find the index of the first negative  number before the peak (where the sign of the signal change)
+    i = peak_index
+    for i in range(peak_index - 1, -1, -1):
+        if arr[i] < 0:
+            return i  # Return the index of the smallest positive number before the peak
+
+    return i
```

Comment:
I think this can be simplified by using `np.where` to look for the indices of all values that are negative and then check for the first that is smaller then the peak.

Also in the edgecase, that there are no negative values, do you really want to return i? Or should be return np.nan, as no proper TC can be indetified.